Controls/Grid/Grid.UWP/View/Services/Selection/SelectionService.cs

```diff
@@ -540,11 +540,18 @@ private void SelectRangeUnits(int row, int startColumnIndex, int endColumnIndex)
 
         private void SelectRangeCells(int row, int column = -1)
         {
-            var cells = this.Owner.Model.CellsController.GetCellsForRow(row);
-            GridCellModel model = column != -1 ? cells.FirstOrDefault(a => a.Column.ItemInfo.Slot == column) : cells.FirstOrDefault();
-            if (model != null)
+            var item = this.Owner.GetDataView().Items[row];
```

Comment:
"this.Owner" is invoked several times inside the method. Lets assign it to a local variable "var owner = this.Owner" and use it instead of invoking the getter of the Owner property when we need it.

Reply:
Thanks @APopatanasov. I'll work on another iteration of this with your guidance. You're correct that I didn't consider grouping, which was a pretty big oversight.